tooling/jbang/ReactiveTestTemplate.java.qute

```diff
@@ -0,0 +1,250 @@
+//DEPS org.assertj:assertj-core:3.19.0
+//DEPS junit:junit:4.13.2
+//DEPS org.testcontainers:postgresql:1.15.3
+//DEPS org.testcontainers:mysql:1.15.3
+//DEPS org.testcontainers:db2:1.15.3
+//DEPS org.testcontainers:mariadb:1.15.3
+//DEPS org.testcontainers:cockroachdb:1.15.3
+//DEPS org.slf4j:slf4j-simple:1.7.30
+//// Testcontainer needs the JDBC drivers to start the containers
+//// Hibernate Reactive doesn't use them
+//DEPS org.postgresql:postgresql:42.2.16
+//DEPS mysql:mysql-connector-java:8.0.25
+//DEPS org.mariadb.jdbc:mariadb-java-client:2.7.3
+//
+
+import java.util.function.Supplier;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+
+import org.hibernate.boot.registry.StandardServiceRegistry;
+import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
+import org.hibernate.cfg.Configuration;
+import org.hibernate.reactive.mutiny.Mutiny;
+import org.hibernate.reactive.provider.ReactiveServiceRegistryBuilder;
+import org.hibernate.reactive.provider.Settings;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.JUnitCore;
+import org.junit.runner.Result;
+import org.junit.runner.RunWith;
+import org.junit.runner.notification.Failure;
+
+import io.vertx.ext.unit.Async;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+import org.testcontainers.containers.CockroachContainer;
+import org.testcontainers.containers.Db2Container;
+import org.testcontainers.containers.JdbcDatabaseContainer;
+import org.testcontainers.containers.MariaDBContainer;
+import org.testcontainers.containers.MySQLContainer;
+import org.testcontainers.containers.PostgreSQLContainer;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+//DESCRIPTION An example of a JUnit test class for Hibernate Reactive using
+//DESCRIPTION [Vert.x Unit](https://vertx.io/docs/vertx-unit/java) and
+//DESCRIPTION [Testcontainers](https://www.testcontainers.org)
+//DESCRIPTION that you can run using [JBang](JBang).
+//DESCRIPTION 
+//DESCRIPTION Before running the tests, Testcontainers will start the selected
+//DESCRIPTION Docker image with the required database created.
+//DESCRIPTION 
+//DESCRIPTION The `DATABASE` constant define which database to use and
+//DESCRIPTION it can be change to any of the values in `Database`.
+//DESCRIPTION 
+//DESCRIPTION Usage example:
+//DESCRIPTION   1. Use as jbang template `jbang init -t reproducer@hibernate/hibernate-reactive mytest.java`
+//DESCRIPTION   2. Run the test with JBang: `jbang mytest.java`
+//DESCRIPTION   3. (Optional) Edit the file (with IntelliJ IDEA for example):
+//DESCRIPTION             jbang edit --live --open=idea SampleIssueTest.java
+@RunWith(VertxUnitRunner.class)
+public class {baseName} {
+
+	/**
+	 * The database to use for the tests
+	 */
+	public final static Database DATABASE = Database.POSTGRESQL;
```

Comment:
Out of curiosity, is there a way to pass the database as a parameter during the template generation?
So that I could replace this with something like:
```
public final static Database DATABASE = Database.{database};
```

and maybe only include the dependencies for the selected database.

I'm trying to avoid a single template for each database but maybe that's the way to go.

Reply:
you *should* be able to refer to -Dkey=value in the templates but I haven't really tested/explored it yet. i've been waiting for a usecase to make the templates "smarter" - this one might be it.

feel free opne jbang issue on it and lets see - I'm cautious to avoid too much logic capbailities to the templating as I also want to be sure the templates generates even if no templates specified.